sqlmesh/dbt/adapter.py

```diff
@@ -219,7 +219,14 @@ def list_relations(self, database: t.Optional[str], schema: str) -> t.List[BaseR
         reference_relation = self.relation_type.create(
             database=database, schema=schema, quote_policy=self.quote_policy
         )
-        return self.list_relations_without_caching(reference_relation)
+        normalized_schema = self._normalize(self._schema(reference_relation))
+
+        relations = self.cache.get(normalized_schema)
```

Comment:
I might suggest making the `CachingAdapter` an owner of the `cache` and expose an API to get it

Reply:
That's right! This was my intention all along, I forgot to update the constructor

Reply:
I will probably update the logic to move `self.cache` in the caching adapter.